ios/brave-ios/Sources/DataImporter/UI/DataImporterView.swift

```diff
@@ -0,0 +1,361 @@
+    self.onDismiss = onDismiss
+  }
+
+  public var body: some View {
+    ZStack {
+      if model.importState == .success {
+        DataImporterStateView(kind: .success, model: model) {
+          model.removeZipFile()
+
+          onDismiss()
+          dismiss()
+        }
+        .padding(16.0)
+        .frame(maxWidth: .infinity, maxHeight: .infinity)
+        .background(Color(braveSystemName: .pageBackground))
+        .toolbar(.hidden, for: .navigationBar)
+        .opacity(showSuccessView ? 1.0 : 0.0)
+        .onAppear {
+          withAnimation(.easeInOut(duration: 0.5)) {
+            showSuccessView = true
+          }
+        }
+        .onDisappear {
+          withAnimation(.easeInOut(duration: 0.5)) {
+            showSuccessView = false
+          }
+        }
+      } else if model.importState == .failure {
+        DataImporterStateView(kind: .failure, model: model) {
+          model.removeZipFile()
+          model.resetAllStates()
+        }
+        .padding(16.0)
+        .frame(maxWidth: .infinity, maxHeight: .infinity)
+        .background(Color(braveSystemName: .pageBackground))
+        .toolbar(.hidden, for: .navigationBar)
+        .opacity(showFailureView ? 1.0 : 0.0)
+        .onAppear {
+          withAnimation(.easeInOut(duration: 0.5)) {
+            showFailureView = true
+          }
+        }
+        .onDisappear {
+          withAnimation(.easeInOut(duration: 0.5)) {
+            showFailureView = false
+          }
+        }
+      } else if model.importState == .importing {
+        DataImporterLoadingView()
+          .padding(16.0)
+          .frame(maxWidth: .infinity, maxHeight: .infinity)
+          .background(Color(braveSystemName: .pageBackground))
+          .toolbar(.hidden, for: .navigationBar)
+          .opacity(showLoadingView ? 1.0 : 0.0)
+          .onAppear {
+            withAnimation(.easeInOut(duration: 0.5)) {
+              showLoadingView = true
+            }
+          }
+          .onDisappear {
+            withAnimation(.easeInOut(duration: 0.5)) {
+              showLoadingView = false
+            }
+          }
+      } else {
+        GeometryReader { proxy in
+          ScrollView {
+            mainView()
+              .frame(minHeight: proxy.size.height)
+          }
+          .background(Color(braveSystemName: .pageBackground))
+        }
+        .sheet(
+          isPresented: $model.isLoadingProfiles,
+          onDismiss: {
+            model.removeZipFile()
+          },
+          content: {
+            if let zipFileURL = model.zipFileURL {
+              GeometryReader { proxy in
+                ScrollView {
+                  multiProfileView(zipFileURL: zipFileURL)
+                    .frame(minHeight: proxy.size.height)
+                }
+                .osAvailabilityModifiers({
+                  if #available(iOS 16.4, *) {
+                  } else {
+                    $0.background(.thickMaterial)
+                  }
+                })
```

Comment:
As Per DM you can't leave one condition empty, need to return the view provided as the closure is a ViewBuilder